### PR TITLE
[MM-23727] Proposal to remove unnecessarily strict channel url validation

### PR DESCRIPTION
**Pull request by fmunshi**
#### Summary
- This ticket was originally for fixing inconsistent validation when renaming the channel url property (Restricted channel urls from containing `__` on `create` but not on `rename`)

- After our discussion on https://github.com/mattermost/mattermost-server/pull/14230 it was found that restricting a channel url to not contain `__` is not required.
Only channels of the format `UserId__UserId` needed to be prevented from being created for non direct channels (eg. `uzsfmtmniifsjgesce4u7yznyh__uzsfmtmniifsjgesce5u7yznyh`)

- Changes proposed in this PR would change channel URL validation to only ensure that something of the format `UserId__UserId` cannot be saved, resulting in a validation error that looks like `User ids are not permitted in channel URL for non-direct message channels.` (Wording to Require PM / UX Review)

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-23727

#### Screenshots
Screen Shot 2020-04-16 at 4 00 35 PM

### CODE REVIEW

**Comment by fmunshi** [comment]
@andrewbrown00 I have attached some screenshots based on the discussion we had on mattermost
let me know if you notice any issues 
Screen Shot 2020-04-16 at 9 18 44 PM
Screen Shot 2020-04-16 at 9 16 57 PM

Screen Shot 2020-04-16 at 9 09 46 PM
Screen Shot 2020-04-16 at 9 09 37 PM
Screen Shot 2020-04-16 at 9 00 15 PM
Screen Shot 2020-04-16 at 9 00 09 PM

**Comment by fmunshi** [comment]
Updated screenshot with fixed wording for new validation error "Start and end with..." 
Screen Shot 2020-04-17 at 2 05 41 PM

**Comment by andrewbrown00** [comment]
Thanks @fmunshi, I replied in our DM to resolve to items.

**Comment by andrewbrown00** [comment]
Adding the updated messages for visibility based on my review with @fmunshi.

`HELP TEXT` You can use lowercase letters, numbers, dashes, and underscores.
`START WITH` URLs must start with a lowercase letter or number.
`END WITH` URLs must end with a lowercase letter or number.
`START AND END` URLs must start and end with a lowercase letter or number.
`Invalid DM` User IDs are not allowed in channel URLs.
`Invalid URL` URLs must have at least 2 characters.
`Invalid Name` Display name must have at least 2 characters.

**Comment by fmunshi** [comment]
Thanks @andrewbrown00 I updated the error message to address the last couple of issues you mentioned and have attached screenshots showing those two changes. 
Screen Shot 2020-04-20 at 3 26 00 PM
Screen Shot 2020-04-20 at 3 25 23 PM

File: utils/url.tsx [hunk 1]
```diff
@@ -107,20 +107,32 @@ function formattedError(id: string, message: string): React.ReactElement {
 export function validateChannelUrl(url: string): React.ReactElement[] {
     const errors: React.ReactElement[] = [];
 
+    let isDirectMessageFormat = false;
+    if (url.indexOf('__') > -1) {
+        const USER_ID_LENGTH = 26;
+        const userIds = url.split('__');
+        isDirectMessageFormat = userIds.length === 2 && userIds[0].length === USER_ID_LENGTH && userIds[1].length === USER_ID_LENGTH;
```
Comment (hmhealey) [inline utils/url.tsx:114]: 0/5, but you could simplify this whole check into a regex with `/^.{26}__.{26}$/`

Comment (fmunshi) [reply utils/url.tsx]: Done @hmhealey - I still ended up using a constant for `USER_ID_LENGTH` though

**Comment by fmunshi** [comment]
Bump @sudheerDev for dev review

**Comment by fmunshi** [comment]
@stevemudie a UserID is any 26 character string so the format that will return an error on rename will only be two 26 character strings separated by a `__`. I tried it on the test server and the error did get thrown for me.

**Comment by stevemudie** [comment]
@fmunshi I've used both an actual 26 character UserID as well as a random one and even tried using two different 26 character strings all with an underscore between them when renaming a channel and still haven't seen any errors. 
No errors when renaming

**Comment by fmunshi** [comment]
@stevemudie Looks like you dont have two underscores between the user ids
the format is 26 character string => two underscores => 26 character string

**Comment by fmunshi** [comment]
Sorry for the confusion 😅

**Pull request merged**